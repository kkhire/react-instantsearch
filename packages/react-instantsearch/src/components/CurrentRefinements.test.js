import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Connect, { CurrentRefinements } from './CurrentRefinements';

Enzyme.configure({ adapter: new Adapter() });

describe('CurrentRefinements - Raw', () => {
  const defaultProps = {
    items: [],
    cx: (...x) => x.join(' '),
    refine: () => {},
    translate: x => x,
  };

  it('renders a list of current refinements', () => {
    const props = {
      ...defaultProps,
      items: [
        {
          label: 'Genre',
          value: () => {},
        },
      ],
    };

    const wrapper = shallow(<CurrentRefinements {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list of nested current refinements', () => {
    const props = {
      ...defaultProps,
      items: [
        {
          label: 'Genre',
          value: () => {},
          items: [
            {
              label: 'Sci-fi',
              value: () => {},
            },
          ],
        },
      ],
    };

    const wrapper = shallow(<CurrentRefinements {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('call refine onClick', () => {
    const props = {
      ...defaultProps,
      items: [
        {
          label: 'Genre',
          value: () => {},
        },
      ],
      refine: jest.fn(),
    };

    shallow(<CurrentRefinements {...props} />)
      .find('button')
      .simulate('click');

    expect(props.refine).toHaveBeenCalledWith(expect.any(Function));
  });
});

describe('CurrentRefinements - Connect', () => {
  const defaultProps = {
    items: [],
    cx: (...x) => x.join(' '),
    refine: () => {},
  };

  it('renders with translate', () => {
    const props = {
      ...defaultProps,
      items: [
        {
          label: 'Genre',
          value: () => {},
        },
      ],
    };

    const wrapper = shallow(<Connect {...props} />).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
