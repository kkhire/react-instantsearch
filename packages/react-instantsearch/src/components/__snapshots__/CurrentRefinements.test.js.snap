// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentRefinements - Connect renders with translate 1`] = `
<ul
  className="list"
>
  <li
    className="item"
    key="Genre"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <span
        className="label"
      >
        Genre
      </span>
      <span
        className="delete"
      >
        âœ•
      </span>
    </button>
  </li>
</ul>
`;

exports[`CurrentRefinements - Raw renders a list of current refinements 1`] = `
<ul
  className="list"
>
  <li
    className="item"
    key="Genre"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <span
        className="label"
      >
        Genre
      </span>
      <span
        className="delete"
      >
        clearFilter
      </span>
    </button>
  </li>
</ul>
`;

exports[`CurrentRefinements - Raw renders a list of nested current refinements 1`] = `
<ul
  className="list"
>
  <li
    className="item"
    key="Sci-fi"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <span
        className="label"
      >
        Sci-fi
      </span>
      <span
        className="delete"
      >
        clearFilter
      </span>
    </button>
  </li>
</ul>
`;
